concordium-consensus/package.yaml

```diff
@@ -59,6 +59,7 @@ dependencies:
 - recursion-schemes >= 5.1
 - file-embed >= 0.0.11
 - lmdb
+- lrucache
```

Comment:
What evaluation was done of this dependency? Looking at it on hackage, it is not very widely used.

Looking at the code a bit, it does not seem to be written with high performance as a major goal. For example it uses Data.Map which is for most purposes quite a bit slower than HashMap or IntMap (if this applies). It partially also uses the strict map, but in many places it also uses `Map.insert` for a lazy map.

These are not problems on their own, but in order to understand the benefits we need to understand the performance of this implementation. Including the potential for space leaks due to laziness.

Reply:
`LRUCache` is just one of the alternatives and the main reason to add it was the presence of the `lrucache` package (thus a minimal effort). We have not evaluated the performances yet. If its performance is inadequate, we could switch to another cache implementation or perhaps delete it entirely.

Reply:
I don't understand this reasoning. What are you evaluating exactly then?

The main thing we should be considering is the performance of our node and what would make most sense. If we use a library for caching that has poor performance for the specific algorithm then we are not evaluating whether the specific caching strategy would make sense. And that is what we should be doing.

Reply:
Nope, that's not quite right. The current LRU solution might still outperform the others. Let's start simple by using the 3rd party solution and maybe it will be sufficient. if not, we can develop our own solution too. Let's wait and see how the performances turn out. Then we may go into more detail.

Reply:
I would be in favor of not using this dependency because of the reasons already stated above.

We supposedly can do much better using better data structures, `IntMap` instead of the generic `Data.Map` in particular. - We need a cache with performance of very high priority - and I think using this for benchmarks would potentially rise as many questions as it will answers.

I think we should try implement our own LRU with performance in mind.

Reply:
The benchmars have shown that the `lrucache` does not perform very well when compared to our own `FIFOCache` implementation. We'll implement replace it with our own LRUCache implementation.